docs/maintainers/adr/0015-introduction-of-implicit-conversions.md

```diff
@@ -0,0 +1,91 @@
+# Introduction of implicit conversions
+
+* Status: accepted
+* Date: 2021-09-09
+
+Technical Story: PLATUI-1035
+
+## Context and Problem Statement
+
+Frequently occurring usage patterns of play-frontend-hmrc components require repetitious boilerplate
+involving excessively nested case class instantiations. This has driven many teams to create wrappers to 
+simplify the construction of components, which are then copied and pasted from service to service.
+
+For example, in order to add a legend to a GovukRadio the following boilerplate is required:
+
+```scala
+Some(Fieldset(
+ legend = Some(Legend(
+   content = Text("message.key.for.legend"),
+   classes = "govuk-fieldset__legend--l",
+   isPageHeading = true
+ ))
+))
+```
+
+In the above example, the only thing that changes between instantiations is the `message.key.for.legend`, everything
+else remains the same.
+
+Should we solve this problem through the introduction of implicit conversions that will automatically wrap simple
+objects such as Strings with the additional boilerplate necessary for commonly occurring cases?  
+
+## Decision Drivers
+
+* The need for API consistency in play-frontend-hmrc.
+* The mixed sentiments towards implicit conversions in the Scala community and slight risk the feature may be removed 
+  entirely in a future Scala version:
+  * See https://www.rallyhealth.com/coding/implicit-implications-part-2-implicit-conversions,
+    https://contributors.scala-lang.org/t/can-we-wean-scala-off-implicit-conversions/4388
+* The risk of unintended side effects if these conversions are added unilaterally 
+  without service developers needing to specifically opt in to them.
+* The repeated invocation of the `Messages` apply method to populate component content parameters, e.g. 
+  Hint, Key, Label, Legend.
+* The difficulty of debugging code involving implicit conversions without turning on advanced IDE features that
+  show implicit hints.
+
+## Considered Options
+
+* Option 1: tackle via extension methods and helpers without introducing implicit conversions
```

Comment:
How about:
* Option 1: tackle via extension methods using implicit Rich helpers and postfix methods without introducing implicit conversions